### PR TITLE
Added validation for GoQuorum transactions with value.

### PR DESCRIPTION
**Pull request by mark-terry**
Signed-off-by: Mark Terry <mark.terry@consensys.net>

## PR description
Adds validation for the case where a GoQuorum transaction has a value.

## Changelog

- [x] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

File: ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/TransactionPool.java [hunk 1]
```diff
@@ -276,6 +285,11 @@ private BlockHeader getChainHeadBlockHeader() {
     return blockchain.getBlockHeader(blockchain.getChainHeadHash()).get();
   }
 
+  private boolean isQuorumTransaction(final Transaction transaction) {
```
Comment (macfarla) [inline ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/TransactionPool.java:288]: isGoQuorumTransaction

File: ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/TransactionPool.java [hunk 2]
```diff
@@ -232,6 +233,14 @@ public PendingTransactions getPendingTransactions() {
   private ValidationResult<TransactionInvalidReason> validateTransaction(
       final Transaction transaction) {
     final BlockHeader chainHeadBlockHeader = getChainHeadBlockHeader();
+
+    // Check whether it's a GoQuorum transaction
+    if (isQuorumTransaction(transaction)) {
+      if (!transaction.getValue().isZero()) {
```
Comment (pinges) [inline ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/TransactionPool.java:239]: I think it is possible that getValue() returns null. Better check that!

File: ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/TransactionPool.java (hunk 1 above)
Comment (pinges) [inline ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/TransactionPool.java:288]: isQuorumPrivateTransaction

**Review by macfarla — APPROVED** [review]
minor comment

File: ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/TransactionPool.java [hunk 3]
```diff
@@ -276,6 +285,11 @@ private BlockHeader getChainHeadBlockHeader() {
     return blockchain.getBlockHeader(blockchain.getChainHeadHash()).get();
   }
 
+  private boolean isQuorumTransaction(final Transaction transaction) {
+    return (transaction.getV().compareTo(BigInteger.valueOf(37)) == 0)
```
Comment (pinges) [inline ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/TransactionPool.java:289]: I think using BigInteger.equals is cheaper!

**Review by pinges — APPROVED** [review]
Please check my comments

**Pull request merged**